### PR TITLE
Event creation

### PR DESCRIPTION
**Pull request by margarita-sh**

### CODE REVIEW

File: backend/src/components/Events/model.ts [hunk 1]
```diff
@@ -11,7 +11,8 @@ export interface IEventsnModel extends Document {
     maxScore: number;
     tasks: Types.ObjectId[];
     users: Types.ObjectId[];
-    participations: Types.ObjectId[];
+	participations: Types.ObjectId[];
```
Comment (maximwow70) [inline backend/src/components/Events/model.ts:14]: Looks like need to sync breaks.
image

File: frontend/src/app/app-routing.module.ts [hunk 2]
```diff
@@ -53,6 +53,13 @@ export const routes: Routes = [
 				(m: typeof import('./libs/common-components-demo/common-components-demo.module')) => m.CommonComponentsDemoModule
 			),
 	},
+	{
+		path: 'eventcreation',
```
Comment (maximwow70) [inline frontend/src/app/app-routing.module.ts:57]: Path is visible for user, so, can we do it more prettier?
Like **create-event**.

File: frontend/src/app/libs/common-components/components/breadcrumb/breadcrumb.component.html [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<div class="breadcrumb">
```
Comment (maximwow70) [inline frontend/src/app/libs/common-components/components/breadcrumb/breadcrumb.component.html:1]: It looks strange when **breadcrumb** component render **breadcrums**. 
My suggestion to name it maybe **breadcrumb-list** or **breadcrumb-panel**, smth like that.
Also, breadcrumb template can be placed in breadcrumb component (it's not necessary, can do it in separated task with refactoring).

File: frontend/src/app/libs/common-components/components/ckeditor/ckeditor.component.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+			{ name: 'styles', items: ['Font', 'Format', 'FontSize'] },
+			{ name: 'basicstyles', groups: ['basicstyles', 'cleanup'], items: ['Bold', 'Italic', 'Underline', 'Strike', 'Subscript', 'Superscript', '-', 'CopyFormatting', 'RemoveFormat'] },
+			{ name: 'paragraph', groups: ['list', 'indent', 'blocks', 'align', 'bidi'], items: ['NumberedList', 'BulletedList', '-', 'Outdent', 'Indent', '-', 'Blockquote', 'CreateDiv', '-', 'JustifyLeft', 'JustifyCenter', 'JustifyRight', 'JustifyBlock', '-', 'BidiLtr', 'BidiRtl', 'Language'] },
+			{ name: 'links', items: ['Link', 'Unlink', 'Anchor'] },
+			{ name: 'insert', items: ['Image', 'Flash', 'Table', 'HorizontalRule', 'Smiley', 'SpecialChar', 'PageBreak', 'Iframe'] },
+			{ name: 'colors', items: ['TextColor', 'BGColor'] },
+			{ name: 'tools', items: ['Maximize', 'ShowBlocks'] }
+		]
+	};
+	theme: any;
```
Comment (maximwow70) [inline frontend/src/app/libs/common-components/components/ckeditor/ckeditor.component.ts:45]: Need to add **public** here.

File: frontend/src/app/libs/common-components/components/ckeditor/ckeditor.component.ts [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+	public constructor(
+		@Optional() @Host() @SkipSelf() protected parentFormContainer: ControlContainer,
+		public elementRef: ElementRef,
+	) { }
+
+	public ngOnInit(): void {
+		this.formControlName = this.elementRef.nativeElement.getAttribute('formControlName');
+		if (this.formControlName != null && this.parentFormContainer != null) {
```
Comment (maximwow70) [inline frontend/src/app/libs/common-components/components/ckeditor/ckeditor.component.ts:54]: Not strict equal is deprecated in usage, please, use type conversion and check on null.
Or in this case I suppose `Boolean(this.formControlName) && ..` will be enough.

File: frontend/src/app/libs/common-components/components/ckeditor/ckeditor.component.ts [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+	public onValueChange(value: any): void {
+		this.onTouched();
+		if (Boolean(this.value) && this.value !== value) {
+			this.setValue(value);
+			this.valueChange.emit(value);
+		}
+
+	}
+	public ngAfterViewInit(): void {
```
Comment (maximwow70) [inline frontend/src/app/libs/common-components/components/ckeditor/ckeditor.component.ts:82]: It's a good practice to place component lifecycle methods in one place after constructor with strict order.

File: frontend/src/app/libs/common-components/components/ckeditor/ckeditor.component.ts [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+		}, 0);
+	}
+
+	public onTouched: any = () => this.value;
+
+	public onChange: any = () => this.value;
+
+	registerOnChange(fn: any): void { this.onChange = fn; }
+
+	registerOnTouched(fn: any): void { this.onTouched = fn; }
```
Comment (maximwow70) [inline frontend/src/app/libs/common-components/components/ckeditor/ckeditor.component.ts:96]: I suppose we need `public` here. What do you think?

File: frontend/src/app/libs/common-components/components/datepicker/datepicker.component.ts [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+
+	@Output() public valueChange: EventEmitter<number | string | boolean | null | undefined> = new EventEmitter();
+
+	public constructor(
+		@Optional() @Host() @SkipSelf() protected parentFormContainer: ControlContainer,
+		public elementRef: ElementRef,
+	) { }
+
+	public ngOnInit(): void {
+		this.formControlName = this.elementRef.nativeElement.getAttribute('formControlName');
```
Comment (maximwow70) [inline frontend/src/app/libs/common-components/components/datepicker/datepicker.component.ts:41]: Hm, this code like in Ckeditor component.
Is it possible to use polymorphism here?
Maybe we can extend this component from Ckeditor, isn't it?

File: frontend/src/app/libs/common-components/components/select/select.component.html [hunk 9]
```diff
@@ -0,0 +1,16 @@
+<div class="app-select">
+	<mat-form-field appearance="outline"
+					class="app-select__field">
```
Comment (maximwow70) [inline frontend/src/app/libs/common-components/components/select/select.component.html:3]: Looks like something happened with spaces.

File: frontend/src/app/libs/common-components/components/select/select.component.ts [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+	@Input() public value: number | string | null | undefined;
+	@Output() public valueChange: EventEmitter<number | string | boolean | null | undefined> = new EventEmitter();
+
+	public constructor(
+		@Optional() @Host() @SkipSelf() protected parentFormContainer: ControlContainer,
+		public elementRef: ElementRef
+	) { }
+
+	public ngOnInit(): void {
+		this.formControlName = this.elementRef.nativeElement.getAttribute('formControlName');
```
Comment (maximwow70) [inline frontend/src/app/libs/common-components/components/select/select.component.ts:32]: And this component the same.
So, I suppose we need base component for all these.

File: frontend/src/app/libs/common-components/components/slide-toggle/slide-toggle.component.html [hunk 11]
```diff
@@ -1,8 +1,8 @@
-<mat-slide-toggle
-	[checked]="value"
-	(change)="toggle()"
-	color="primary"
-	[class.mat-slide-toggle-bar-dark]="isDarkTheme"
-	[disabled]="disabled">
-		{{title}}
-</mat-slide-toggle>
+<mat-slide-toggle [checked]="value"
+				  (change)="toggle()"
```
Comment (maximwow70) [inline frontend/src/app/libs/common-components/components/slide-toggle/slide-toggle.component.html:2]: The same with spaces.

File: frontend/src/app/libs/common-components/components/slide-toggle/slide-toggle.component.ts [hunk 12]
```diff
@@ -18,18 +26,76 @@ export class SlideToggleComponent {
...
+		}, 0);
+	}
+
+	public onTouched: any = () => this.value;
+
+	public onChange: any = () => this.value;
+
+	registerOnChange(fn: any): void { this.onChange = fn; }
+
+	registerOnTouched(fn: any): void { this.onTouched = fn; }
```
Comment (maximwow70) [inline frontend/src/app/libs/common-components/components/slide-toggle/slide-toggle.component.ts:94]: The same with access modifier.

File: frontend/src/app/libs/common-components/components/select/select.component.html (hunk 9 above)
Comment (clain-atomi) [reply frontend/src/app/libs/common-components/components/select/select.component.html:3]: +

**Review by clain-atomi — APPROVED** [review]

**Pull request merged**